vectorizers/preprocessing.py

```diff
@@ -57,10 +57,10 @@ def construct_timed_document_frequency(token_by_doc_sequence, token_dictionary):
     n_tokens = len(token_dictionary)
     doc_freq = np.zeros(n_tokens)
     for doc in token_by_doc_sequence:
+        doc_set = set([token[0] for token in doc])
         doc_freq += np.bincount(
-            [token_dictionary[token[0]] for token in set(doc)], minlength=n_tokens
+            [token_dictionary[token[0]] for token in doc_set], minlength=n_tokens
```

Comment:
```suggestion
            [token_dictionary[token] for token in doc_set], minlength=n_tokens
```
I think you don't need the ``[0]`` index anymore since you have the ``doc_set`` which already did it.